book/src/appendices/b_migration_notes/cgmath_to_nalgebra.md

```diff
@@ -0,0 +1,228 @@
+        -transform_0.translation - transform_1.translation
+        +transform_0.translation() - transform_1.translation()
+
+        -transform.translation[0] = x;
+        +transform.set_x(position.x);
+
+        -translation.x += 0.1;
+        -translation.y -= 0.1;
+        +translation.translate_x(0.1);
+        +translation.translate_y(-0.1);
+
+        -let ball_x = transform.translation[0];
+        +let ball_x = transform.translation().x;
+
+        -transform.set_position(Vector3::new(6.0, 6.0, -6.0));
+        +transform.set_xyz(6.0, 6.0, -6.0);
+
+        // Rotations
+        -use amethyst::core::cgmath::Deg;
+        -
+        -transform.set_rotation(Deg(75.96), Deg(0.0), Deg(0.0));
+        +transform.rotate_local(Vector3::x_axis(), 1.3257521);
+        ```
+
+    - `amethyst::renderer::Transform` prefabs no longer use labels
+
+        ```patch
+         // scene.ron
+         data: (
+             transform: (
+        -        translation: (x: 0.0, y: 0.0, z: -4.0),
+        -        rotation: (s: 0.0, v: (x: 0.0, y: 1.0, z: 0.0),),
+        -        scale: (x: 4.0, y: 2.0, z: 1.0),
+        +        translation: (0.0, 0.0, -4.0),
+        +        rotation: (0.0, 0.0, 1.0, 0.0),
+        +        scale: (4.0, 2.0, 1.0),
+             ),
+        ```
+
+* `amethyst::renderer::GlobalTransform` inverse.
+
+    ```patch
+    -global.0.invert()
+    +global.0.try_inverse()
+    ```
+
+* `amethyst::renderer::Pos*` fields use `nalgebra` types instead of arrays.
+
+    Type change:
+
+    ```patch
+     pub struct PosTex {
+    -    pub position: [f32; 3],
+    -    pub tex_coord: [f32; 2],
+    +    pub position: Vector3<f32>,
+    +    pub tex_coord: Vector2<f32>,
+     }
+    ```
+
+    Usage changes:
+
+    ```patch
+     PosTex {
+    -    position: [0.0, 0.0, 0.0],
+    -    tex_coord: [0.0, 0.0],
+    +    position: Vector3::new(0.0, 0.0, 0.0),
+    +    tex_coord: Vector2::new(0.0, 0.0),
+     }
+    ```
+
+* `amethyst::core::nalgebra::Matrix4` construction.
+
+    ```patch
+    -Matrix4::from_translation(Vector3::new(x, y, z))
+    +Matrix4::new_translation(&Vector3::new(x, y, z))
+
+    // OR
+
+    +use amethyst::core::nalgebra::Translation3;
+    +
+    +Translation3::new(x, y, z).to_homogeneous()
+    ```
+
+* `UnitQuarternion::rotation_between` is right handed, previously they were left handed.
+* Orthographic projection has changed from `(left, right, top, bottom)` to `(left, right, bottom, top)`.
+
+    ```patch
+     Projection::orthographic(
+         0.0,           // left
+         ARENA_WIDTH,   // right
+    -    ARENA_HEIGHT,  // top
+         0.0,           // bottom
+    +    ARENA_HEIGHT,  // top
+     )
+
+    -use amethyst::core::cgmath::Ortho;
+    -
+    -Ortho { left, right, top, bottom, near, far }
+    +use amethyst::core::nalgebra::Orthographic3;
+    +
+    +Orthographic3::new(left, right, bottom, top, near, far)
+    ```
+
+* Perspective projection
+
+    - Angles are in radians instead of degrees:
```

Comment:
It's not instead but rather just that nalgebra doesn't have degree types. You could still use `Rad()` in cgamth if you wanted to.